File: C#/DeckOfCards/Deck.cs
using System;
using System.Collections.Generic;

namespace DeckOfCards {
    public class Deck {
        public List<Card> cards;

        public Deck(){
            reset();
        }

        public void reset(){
            cards = new List<Card>();
            string[] suits = new string[4] {"Hearts", "Clubs", "Spades", "Diamonds"};
            for(int i = 0; i < suits.Length; i++){
                for(int n = 1; n < 14; n ++) {
                    cards.Add(new Card(suits[i], n));
                }
            }
        }

        public Card deal() {
            Card toReturn = cards[0];
            cards.RemoveAt(0);
            return toReturn;
        }

        public void shuffle(){
            Random rand = new Random();
            for(var i = cards.Count - 1; i > 0; i--) {
                int randIdx = rand.Next(cards.Count - i);
                Card swap = cards[i];
                cards[i] = cards[randIdx];
                cards[randIdx] = swap;
            }
        }

        public override string ToString() {
            string str = "";
            foreach(Card card in cards) {
                str += card + "\n";
            }
            return str;
        }
    }
}

File: Mean/blackjack/deck.js

function Deck() {
  this.deck = []
  this.populateDeck().shuffle().shuffle().shuffle()
}

// adds 52 cards to this.deck[]
Deck.prototype.populateDeck = function() {
  var suites = ['spades', 'hearts', 'diamonds', 'clubs']

  for (var i = 0; i < suites.length; i++) {
    for (var j = 1; j < 14; j++) {
      var card = new Card(suites[i], j)
      this.deck.push(card)
    }
  }

  return this
}

// shuffle the deck, can be chained
Deck.prototype.shuffle = function() {
  if (!this.deck.length) {
    console.log('Deck has not been populated!')
    return -1
  }

  for (var i = 0; i < this.deck.length; i++) {
    var tmp
    var rand = getRandom()
    tmp = this.deck[i]
    this.deck[i] = this.deck[rand]
    this.deck[rand] = tmp
  }

  function getRandom() {
    return Math.floor(Math.random() * (52 - 0) + 0)
  }

  return this
}

// reset
Deck.prototype.reset = function() {
  this.deck = []
  this.populateDeck().shuffle().shuffle().shuffle()
}


// deal
Deck.prototype.deal = function() {
  return this.deck.pop()
}
